BootloaderCorePkg/Library/AcpiInitLib/AcpiInitLib.c

```diff
@@ -392,10 +393,17 @@ UpdateMadt (
   //
   // Copy rest of MADT entries
   //
+
+  if (FeaturePcdGet (PcdCpuX2ApicEnabled)) {
+    ApicStrucType = EFI_ACPI_5_0_PROCESSOR_LOCAL_X2APIC;
+  } else {
+    ApicStrucType = EFI_ACPI_5_0_PROCESSOR_LOCAL_APIC;
+  }
+
   while (MadtPtr < MadtEnd) {
     EntryHeader = (EFI_ACPI_MADT_ENTRY_COMMON_HEADER *)MadtPtr;
     Length = EntryHeader->Length;
-    if (EntryHeader->Type != EFI_ACPI_5_0_PROCESSOR_LOCAL_APIC) {
```

Comment:
How about do minimal change as below?
if ((EntryHeader->Type != EFI_ACPI_5_0_PROCESSOR_LOCAL_APIC) && (EntryHeader->Type != EFI_ACPI_5_0_PROCESSOR_LOCAL_X2APIC)) {

Reply:
Great. The simpler, the better